### PR TITLE
Add Redis password authentication

### LINKED ISSUE #204: Missing password authentication for Redis
I'd like to protect my Redis instance with a password, but unfortunately IcingaDB is not able to connect to Redis using a password.  

```
# vim /etc/icinga-redis/icingadb-redis.conf

requirepass icingadb
```

```
# vim /etc/icingadb/icingadb.ini 

[redis]
host="127.0.0.1"
;port=6380
password="icingadb"
```

This configuration results in this error message:

```
Aug 09 11:11:01 deb10-icinga systemd[1]: Started Icinga DB.
Aug 09 11:11:01 deb10-icinga icingadb[27074]: time="2020-08-09T11:11:01+02:00" level=info msg="Connecting to Redis"
Aug 09 11:11:01 deb10-icinga icingadb[27074]: time="2020-08-09T11:11:01+02:00" level=error msg="Could not connect to Redis. Trying again" context=redis error="NOAUTH Authentication required."
Aug 09 11:11:01 deb10-icinga icingadb[27074]: time="2020-08-09T11:11:01+02:00" level=info msg="Connecting to MySQL"
Aug 09 11:11:01 deb10-icinga icingadb[27074]: time="2020-08-09T11:11:01+02:00" level=info msg="Starting heartbeat listener"
Aug 09 11:11:01 deb10-icinga icingadb[27074]: time="2020-08-09T11:11:01+02:00" level=info msg="SQL connection established"
```

I thought the `password` field would work since I found a reference in the code:

https://github.com/Icinga/icingadb/blob/f8eb69f5e54b2aec61fe5b8e4bbdbf7882781dfb/config/config.go#L19-L25

### PR DESCRIPTION
**Pull request by mcktr**
Adds the missing password authentication.

fixes #204

### CODE REVIEW

**Review by Al2Klimov — CHANGES_REQUESTED** [review]
Please apply go fmt to the files you changed.

**Comment by mcktr** [comment]
@Al2Klimov No clue how this happened, since my GoLand should run `go fmt` automatically. The formatting is now fixed.

File: main.go [hunk 1]
```diff
@@ -5,6 +5,12 @@ package main
 import (
 	"flag"
 	"fmt"
+	"os"
+	"os/signal"
+	"regexp"
+	"sync"
+	"syscall"
+
```
Comment (Al2Klimov) [inline main.go:13]: Does this really comply to go fmt?

**Review by Al2Klimov — COMMENTED** [review]
```
➜  icingadb git:(feature/password-authentication) ✗ go run . -config nopass.ini
INFO[0000] Connecting to Redis
ERRO[0000] Could not connect to Redis. Trying again      context=redis error="NOAUTH Authentication required."
INFO[0000] Connecting to MySQL
INFO[0000] Starting heartbeat listener
INFO[0000] SQL connection established
^CINFO[0003] Shutting down                                 signal=interrupt
➜  icingadb git:(feature/password-authentication) ✗ go run . -config pass.ini
INFO[0000] Connecting to Redis
INFO[0000] Connecting to MySQL
INFO[0000] Redis connection established
INFO[0000] Starting heartbeat listener
INFO[0000] SQL connection established
^CINFO[0006] Shutting down                                 signal=interrupt
➜  icingadb git:(feature/password-authentication) ✗
```

File: main.go (hunk 1 above)
Comment (mcktr) [reply main.go]: As long as there is a empty line between the imports from the stdlib and the other imports, yes. I removed the empty line and `go fmt` sorted it alphabetical. But this does not comply to `goimports` for example.

**Comment by mcktr** [comment]
@Al2Klimov Would be nice if you added more details to your comment, now I have to guess what you are trying to tell me. It would also be good to know your used configuration e.g. your redis config and the icingadb.ini config.

I can't reproduce this behavior.

 **authentication enabled**

redis:

```
# sudo docker run -ti -h redis -p 6379:6379 redis:latest bash

root@redis:/data# redis-server --requirepass icingadb
```

IcingaDB:

```
# cat pass.ini 

[redis]
host="127.0.0.1"
port=6379
password="icingadb"

[mysql]
host="127.0.0.1"
user="icingadb"
password="icingadb"

[logging]
level="info"

[metrics]
#host="127.0.0.1"
#port=8080
```

```
# go run . -config pass.ini 
INFO[0000] Connecting to Redis                          
INFO[0000] Connecting to MySQL                          
ERRO[0000] Could not connect to SQL. Trying again        context=sql error="dial tcp 127.0.0.1:3306: connect: connection refused"
INFO[0000] Redis connection established                 
INFO[0000] Starting heartbeat listener                  
^CINFO[0029] Shutting down                                 signal=interrupt
```
> Ignore the SQL error, I don't have an MySQL server running locally. This is okay since we are focusing on the redis error. 

The redis connection successfully established

**no authentication enabled**

redis:

```
# sudo docker run -ti -h redis -p 6379:6379 redis:latest bash

root@redis:/data# redis-server
```

IcingaDB:

```
# cat nopass.ini

[redis]
host="127.0.0.1"
port=6379
;password="icingadb"

[mysql]
host="127.0.0.1"
user="icingadb"
password="icingadb"

[logging]
level="info"

[metrics]
#host="127.0.0.1"
#port=8080
```

```
# go run . -config nopass.ini 
INFO[0000] Connecting to Redis                          
INFO[0000] Connecting to MySQL                          
ERRO[0000] Could not connect to SQL. Trying again        context=sql error="dial tcp 127.0.0.1:3306: connect: connection refused"
INFO[0000] Redis connection established                 
INFO[0000] Starting heartbeat listener                  

```
> Ignore the SQL error, I don't have an MySQL server running locally. This is okay since we are focusing on the redis error.

The redis connection successfully established

**Review by Al2Klimov — APPROVED** [review]

**Review by N-o-X — APPROVED** [review]

**Pull request merged**